File: implementations/node/sdk/src/contracts/publication/pipeline-publication-request.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkPipelineDefinition } from "../pipelines/pipeline-definition.js";
import type { AiSdkPublicationFunctionUpload } from "./publication-function-upload.js";

export interface AiSdkPipelinePublicationRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.pipelinePublicationRequest;
  readonly definition: AiSdkPipelineDefinition;
  readonly functions?: readonly AiSdkPublicationFunctionUpload[];
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/publication/pipeline_publication_request.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from ..pipelines.pipeline_definition import AiSdkPipelineDefinition
from .publication_function_upload import AiSdkPublicationFunctionUpload


@dataclass(frozen=True)
class AiSdkPipelinePublicationRequest(AiSdkWireModel):
    definition: AiSdkPipelineDefinition
    functions: tuple[AiSdkPublicationFunctionUpload, ...] = ()
    schema_version: int = AiSdkSchemaVersions.PIPELINE_PUBLICATION_REQUEST
